File: angular-app/src/app/inventory-management/models/process/process-update.ts
import { ProcessState } from 'src/app/common/process-state.enum';
import { UserQuery } from '../user/user-query';

export class ProcessUpdate {
  id!: string | null;
  name!: string;
  description!: string;
  typeId!: number;
  publishingDate!: Date;
  unpublishingDate!: Date;
  usersDataSourceId!: string;
  users: UserQuery[] = [];
  steps: string[] = [];
}

File: dotnet-restapi/InventoryManagementApi/DTOs/Process/ProcessUpdate.cs
using InventoryManagementApi.DTOs.User;
using InventoryManagementApi.Models;
using System;
using System.Collections.Generic;

namespace InventoryManagementApi.DTOs.Process
{
    public class ProcessUpdate
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public int TypeId { get; set; }
        public DateTime? PublishingDate { get; set; }
        public DateTime? UnpublishingDate { get; set; }
        public Guid? UsersDataSourceId { get; set; }
        public ICollection<UserQuery> Users { get; set; }
        public ICollection<Guid> Steps { get; set; }
    }
}
